### DISCUSSION TITLE
How does the Verify User Account method work?

### DISCUSSION THREAD

**Discussion by phil-hudson** [GodotNuts/GodotFirebase#344] [Q&A] [answered]
Hey!

I'm looking for clarification on the method here: https://github.com/GodotNuts/GodotFirebase/wiki/Authentication-and-User-Management#verify-user-account

1) Do we trigger this straight after successful signup? Is there a signal fired when the user successfully verifies? (I haven't seen one in code). I've tried doing this and have not received an email. Would be curious as to how you are using it.
2) How do we prevent unverified users from logging in?

Thanks

**Comment by WolfgangSenff** [comment, marked as answer]
Yes, you would want to trigger this after a successful signup - probably want to switch your view to tell them an email has been sent so that they can verify their email address. There's no signal or anything like that - nothing like that comes back from Firebase. Instead, there's a boolean in their info that says whether or not they're verified. You can see where that comes into play here: https://github.com/GodotNuts/GodotFirebase/blob/a0c55125c3643c53c7888ded7103352f36e23318/addons/godot-firebase/auth/user_data.gd#L12

So for number 2, given the above, what you want to do is "let" them login, but then check that email verified value, and if it's false, immediately log them out. You would want to not show them the game/anything until it's verified, so in theory you should not have navigated by this point. I wish there was an easier way, but if there is, I don't know it. Going to close with this comment, but feel free to re-open if you still want to talk about it.

**Comment by phil-hudson** [comment]
Thanks so much, this is a fantastic answer! Might be work bringing it into docs :)

**Reply by WolfgangSenff** [reply]
I think that's a good idea! I'll see about it as soon as I can.

**Reply by phil-hudson** [reply]
Sorry, also one last question - is there any prerequisite setup required to get emails sending in firebase?

**Reply by WolfgangSenff** [reply]
Not that I'm aware - you should just have to call that function, and whatever email the person tried to register should get one.
